### PR TITLE
Add logging handler

### PR DESCRIPTION
**Pull request by calebkiage**

### CODE REVIEW

File: src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        if (!response.IsSuccessStatusCode)
+        {
+            logger?.LogDebug("{0}\t{1}\t{2}", request.RequestUri,
+                (int)response.StatusCode, response.Headers.Date);
+            logger?.LogDebug("{0}", await response.Content?.ReadAsStringAsync());
+        }
+        return response;
+    }
+
+    public void SetLogger(ILogger<LoggingHandler> logger)
```
Comment (baywet) [inline src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs:39]: any opposition to make the field an auto-property and removing the setter?

**Review by baywet — CHANGES_REQUESTED** [review]

File: src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs (hunk 1 above)
Comment (calebkiage) [reply src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs:39]: Logger shouldn't be accessible outside the class. IIRC, auto-properties must have a getter

Comment (baywet) [reply src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs:39]: public Type Foo { private get; set; }

Comment (calebkiage) [reply src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs:39]: This was the idea I had too. It didn't work. Read access to the logger was still possible outside the class.

File: src/sample/Program.cs [hunk 2]
```diff
@@ -64,14 +66,25 @@ static async Task<int> Main(string[] args)
...
+            using var httpClient = GraphCliClientFactory.GetDefaultClient(options, lowestPriorityMiddlewares: new[] { loggingHandler });
             var core = new HttpClientRequestAdapter(authProvider, httpClient: httpClient);
             commands.Add(UsersCommandBuilder.BuildUsersCommand(core));
 
             var builder = BuildCommandLine(commands).UseDefaults().UseHost(CreateHostBuilder);
             builder.AddMiddleware((invocation) =>
             {
                 var host = invocation.GetHost();
+                var debug = invocation.Parser.Configuration.RootCommand.Options.SingleOrDefault(o => o.Name == "debug") as Option<bool>;
+                var isDebug = debug != null ? invocation.ParseResult.GetValueForOption(debug) : false;
```
Comment (baywet) [inline src/sample/Program.cs:79]:
```suggestion
                var isDebug = invocation.Parser.Configuration.RootCommand.Options.SingleOrDefault(static o => "debug".Equals(o.Name, StringComparison.OrdinalIgnoreCase)) is Option<bool> debug ? invocation.ParseResult.GetValueForOption(debug) : false;
```

**Review by baywet — CHANGES_REQUESTED** [review]

**Review by baywet — APPROVED** [review]

**Pull request merged**